jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/AppServerAutoDetectionTest.java

```diff
@@ -121,28 +122,28 @@ public void testDefaultServer() throws IOException {
             .build()).build();
 
         AppServerHandler appServerHandler = new AppServerDetector(generatorContext).detect(null);
-        assertEquals("tomcat", appServerHandler.getName());
+        assertThat(appServerHandler.getName()).isEqualTo("tomcat");
     }
 
     private void assertAppServerDescriptorApplicability(Object[] descriptors) throws IOException {
         for (int i = 0; i < descriptors.length; i += 2) {
             String descriptor = (String) descriptors[i];
+            String appDir = descriptor.split("/")[0];
+            String file = descriptor.split("/")[1];
             boolean expected = (boolean) descriptors[i + 1];
 
-            File appDir = folder.newFolder("webapp" + i);
-            assertTrue(new File(appDir, "META-INF/").mkdirs());
-            assertTrue(new File(appDir, "WEB-INF/").mkdirs());
-            assertTrue(new File(appDir, descriptor).createNewFile());
+            Path webInf = Files.createDirectories(folder.resolve("webapp" + i).resolve(appDir));
```

Comment:
This path is not resolving to `webapp$i/META-INF/WEB-INF`
Considering L139(L138), we need to keep the other `appDir` variable too.

https://github.com/eclipse/jkube/pull/1691#discussion_r937446403

Reply:
On debugging this, I found out that on each iteration it is creating a directory `webapp$i` and inside it creates two different directories `META-INF` and `WEB-INF`, not nested as you said `webapp$i/META-INF/WEB-INF`
After resolving these different directories it resolves `descriptor` accordingly.

If I do like [#1691 (comment)](https://github.com/eclipse/jkube/pull/1691#discussion_r937446403) then it gives
```java
java.nio.file.NoSuchFileException: /tmp/junit3247740821297201882/webapp0/META-INF/WEB-INF/WEB-INF/jboss-deployment-structure.xml
```
I'm not sure why it's giving error may be `jboss-deployment-structure.xml` being resolved in `webapp0/META-INF/WEB-INF/WEB-INF` (nested WEB-INF) instead of `webapp0/WEB-INF` 

but on doing like 
```java
String appDir = descriptor.split("/")[0];
String file = descriptor.split("/")[1];

Path webInf = Files.createDirectories(folder.resolve("webapp" + i).resolve(appDir));
Files.createFile(webInf.resolve(file));
```
it works

Reply:
The https://github.com/eclipse/jkube/pull/1691#discussion_r937446403 code suggestion is wrong.
For some reason I assumed the directories were nested (probably too many iterations over the same issue).
The test should be migrated to keep the previous behavior but using the nio Files utils instead.